### PR TITLE
Bobs stuff

### PR DESCRIPTION
**Pull request by bobelly**
Hey-hey. Here comes the first of a broad line of amazing future commits that I plan to do ;) In there you will find the promised ReZero and Dynet (names are not my idea). Have fun with it and remember - everything is good

### CODE REVIEW

File: torchsupport/modules/rezero.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class ReZero(nn.Module):
+  r'''Implemets ReZero normalization proposed by Bachlechner et al. 
+    (https://arxiv.org/pdf/2003.04887.pdf) 
+  Args: 
+    function (callable): function you want to use (for example conv2D)'''
+  def __init__(self, function):
+    super().__init__()
+    self.function = function
+    self.alpha = nn.Parameter(torch.tensor(
+      0, dtype=torch.float, requires_grad=True
```
Comment (sholderbach) [inline torchsupport/modules/rezero.py]: Sounds cool! From the paper I'm not sure if their alpha is a single scalar per layer or per channel (as popular with normalizations)

**Review by sholderbach — COMMENTED** [review]
Brief brain-fart if channel-wise ReZero makes sense. In a repo by the authors it seems, to be a layer level scalar (https://github.com/majumderb/rezero)

File: torchsupport/modules/dynet.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as func
+
+class DyNet(nn.Module):
```
Comment (mjendrusch) [inline torchsupport/modules/dynet.py]: Cool, I like it. A nice addition would be to have a DyNet Module for 1d and 3d convolution as well.

File: torchsupport/modules/rezero.py (hunk 1 above)
Comment (mjendrusch) [reply torchsupport/modules/rezero.py]: It seems its a single scalar, but nothing stops us from adding an optional argument with a size for the alpha which defaults to a scalar. Sounds like a natural extension if you ask me.

**Review by mjendrusch — CHANGES_REQUESTED** [review]
It should be straightforward to add separate 1d/2d/3d convolutions, as well as a parameter specifying the alpha size for the ReZero Module. These should make a nice addition for the library.

**Review by mjendrusch — APPROVED** [review]

**Comment by mjendrusch** [comment]
Looks great, thanks!
